src/slicegpt/hf_utils.py

```diff
@@ -173,3 +180,51 @@ def load_sliced_model(
     model_adapter.model.eval()
 
     return model_adapter, tokenizer
+
+
+def save_sliced_model(
+    model_name: str,
+    dtype: torch.dtype,
+    model: torch.nn.Module,
+    scheduler: SlicingScheduler,
+    save_sliced_model_dir: str | pathlib.Path,
+    sparsity: float,
+    new_hidden_size: int,
+    slicing_conf: SlicingConfig,
+):
+    if model_name == "microsoft/phi-2":
+        config = PhiConfig.from_pretrained(
+            model_name,
+            torch_dtype=torch.float16,
+        )
+
+        config.save_pretrained("phi_config")
+        config_to_save = SlicedPhi2Config.from_pretrained(
+            config_path="phi_config", sparsity=sparsity, new_hidden_size=new_hidden_size
```

Comment:
I am curious how `config_path` works here? Is this a real path? Is it not a name of config recognized by HF or a local path? Or is it simply reading back the file saved in the line above? In which case it could, in principle, be saved anywhere, rather than the current directory, right? (Not asking for a change, rather seeking an explanation)

Reply:
Here it's simply reading the file saved above, so we can save it anywhere. Although I think this way of doing this is not ideal. Another option is to copy over each parameter from that config to the new one but it's also not very convenient.  Ideally we want to have all the parameters initialized correctly by default in the config and only pass `sparsity `and `new_hidden_size`. I will think about how to do this better in a subsequent PR.